src/setfit/data.py

```diff
@@ -15,69 +16,124 @@
+        reference_dataset (`str`, *optional*): A dataset to take labels
+        from, if `candidate_labels` is not supplied.
+        template (`str`, *optional*, defaults to `"This sentence is {}"`): The template
+            used to turn each label into a synthetic training example. This template
+            must include a {} for the candidate label to be inserted into the template.
+            For example, the default template is "This sentence is {}." With the
+            candidate label "sports", this would produce an example
+            "This sentence is sports".
+        sample_size (`int`, *optional*, defaults to 2): The number of examples to make for
+        each candidate label.
+        text_column (`str`, *optional*, defaults to `"text"`): The name of the column
+        containing the text of the examples.
+        label_column (`str`, *optional*, defaults to `"label"`): The name of the column
+        in `dataset` containing the labels of the examples.
+        multi_label (`bool`, *optional*, defaults to `False`): Whether or not multiple
+        candidate labels can be true.
+        label_names_column (`str`, *optional*, defaults to "label_text"): The name of the
+        label column in the `reference_dataset`, to be used in case there is no ClassLabel
+        feature for the label column.
+
+    Returns:
+        `Dataset`: A copy of the input Dataset with templated examples added.
+
+    Raises:
+        `ValueError`: If the input Dataset is not empty and one or both of the
+            provided column names are missing.
+    """
+    if dataset is None:
+        dataset = Dataset.from_dict({})
+
+    required_columns = {text_column, label_column}
+    column_names = set(dataset.column_names)
+    if column_names:
+        missing_columns = required_columns.difference(column_names)
+        if missing_columns:
+            raise ValueError(f"The following columns are missing from the input dataset: {missing_columns}.")
+
+    if bool(reference_dataset) == bool(candidate_labels):
+        raise ValueError(
+            "Must supply exactly one of `reference_dataset` or `candidate_labels` to `get_templated_dataset()`!"
+        )
+
+    if candidate_labels is None:
+        candidate_labels = get_candidate_labels(reference_dataset, label_names_column)
+
+    empty_label_vector = [0] * len(candidate_labels)
+
+    for label_id, label_name in enumerate(candidate_labels):
+        label_vector = empty_label_vector.copy()
+        label_vector[label_id] = 1
+        example = {
+            text_column: template.format(label_name),
+            label_column: label_vector if multi_label else label_id,
         }
-    else:
-        raise ValueError(f"Dataset {dataset} not supported for data augmentation!")
+        for _ in range(sample_size):
+            dataset = dataset.add_item(example)
+
+    return dataset
+
+
+def add_templated_examples(**kwargs) -> None:
```

Comment:
```suggestion
def add_templated_examples(*args, **kwargs) -> None:
```
I think there's an error if users call `add_templated_examples(dataset)` otherwise.

Reply:
thanks